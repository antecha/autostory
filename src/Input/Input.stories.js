import React from 'react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, number } from '@storybook/addon-knobs';
import Input from './index';

export default {
  component: Input,
  title: 'Input',
  decorators: [withKnobs],
};

export const InputText = () => <Input name={text('Name', 'username')} value={text('value', 'hey')} />;
