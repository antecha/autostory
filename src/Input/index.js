// @flow
import React from 'react';
import type { Element } from 'react';

type InputProps = { type: 'text' | 'number', name: string, value: string | number };

const Input = ({ type = 'text', name = '', value }: InputProps): Element<'input'> => (
  <input name={name} type={type} value={value} />
);

export default Input;
