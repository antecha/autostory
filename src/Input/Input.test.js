import React from 'react';
import { create } from 'react-test-renderer';
import Input from './index';

describe('Input', () => {
  const testRenderer = (name, type) => create(<Input type={type} name={name} />);

  it('renders', () => {
    expect(testRenderer('username', 'text').toJSON()).toMatchSnapshot();
  });
});
