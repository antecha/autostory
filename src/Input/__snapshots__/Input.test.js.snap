// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders 1`] = `
<input
  name="username"
  type="text"
/>
`;
